src/test/java/CommandDataTest.java

```diff
@@ -35,15 +37,17 @@
     public void testNormal()
     {
         CommandData command = new CommandDataImpl("ban", "Ban a user from this server")
-                .setDefaultEnabled(false)
+                .setGuildOnly(true)
+                .setDefaultPermissions(DefaultMemberPermissions.enabledFor(Permission.BAN_MEMBERS))
                 .addOption(OptionType.USER, "user", "The user to ban", true) // required before non-required
                 .addOption(OptionType.STRING, "reason", "The ban reason") // test that default is false
                 .addOption(OptionType.INTEGER, "days", "The duration of the ban", false); // test with explicit false
 
         DataObject data = command.toData();
         Assertions.assertEquals("ban", data.getString("name"));
         Assertions.assertEquals("Ban a user from this server", data.getString("description"));
-        Assertions.assertFalse(data.getBoolean("default_permission"));
+        Assertions.assertFalse(data.getBoolean("dm_permission"));
+        Assertions.assertEquals(Permission.BAN_MEMBERS.getRawValue(), data.getUnsignedLong("default_member_permissions"));
```

Comment:
We probably should have a test that checks the odd behavior around `null` and `0` too honestly.